openmc/model/funcs.py

```diff
@@ -111,6 +112,151 @@ def borated_water(boron_ppm, temperature=293., pressure=0.1013, temp_unit='K',
+                      ax2 + '0': val2,
+                      'r': r})
+
+    if axis == 'x':
+        x1, x2 = 'y', 'z'
+        axcoord, axcoord1, axcoord2 = 0, 1, 2
+    elif axis == 'y':
+        x1, x2 = 'x', 'z'
+        axcoord, axcoord1, axcoord2 = 1, 0, 2
+    else:
+        x1, x2 = 'x', 'y'
+        axcoord, axcoord1, axcoord2 = 2, 0, 1
+
+    # Create cylindrical region
+    min_h = plane(axis, 'minimum', -height / 2 + origin[axcoord])
+    max_h = plane(axis, 'maximum', height / 2 + origin[axcoord])
+    radial = cylinder(axis, 'outer',
+                      x1, origin[axcoord1],
+                      x2, origin[axcoord2],
+                      r)
+
+    if boundary_type == 'periodic':
+        min_h.periodic_surface = max_h
+        max_h.periodic_surface = min_h
+    prism = +min_h & -max_h & -radial
+
+    # Handle rounded corners if given
+    if upper_fillet_radius > 0. or lower_fillet_radius > 0.:
+        # Get torus class corresponding to given axis
+        tor = globals()['{}Torus'.format(axis.upper())]
+
+        if boundary_type == 'periodic':
+            raise ValueError('Periodic boundary conditions not permitted when '
+                             'rounded corners are used.')
+
+        corners_upper = None
+        corners_lower = None
+        args = {
+            'boundary_type': boundary_type,
+            x1 + '0': origin[axcoord1],
+            x2 + '0': origin[axcoord2]
+        }
+
+        def create_fillet(args, r, fillet_radius, pos):
+            args['a'] = r - fillet_radius
+            args['b'] = fillet_radius
+            args['c'] = fillet_radius
+            if pos:
+                coord = origin[axcoord] + height / 2 - fillet_radius
+                tor_name = f'{axis} max'
+                cyl_name = 'upper_min'
+                axial_bound = +plane(axis, 'upper min', coord)
+            else:
+                coord = origin[axcoord] - height / 2 + lower_fillet_radius
+                tor_name = f'{axis} min'
+                cyl_name = 'lower_min'
+                axial_bound = -plane(axis, 'lower max', coord)
+
+            cyl = cylinder(axis, cyl_name,
+                           x1, origin[axcoord1],
+                           x2, origin[axcoord2],
+                           r - fillet_radius)
+
+            args[axis + '0'] = coord
+            torus = tor(name=tor_name, **args)
+
+            fillet = +cyl & +torus & axial_bound
+            return fillet
+
+        if upper_fillet_radius > 0.:
+            fillet_upper = create_fillet(args, r, upper_fillet_radius, True)
+        if lower_fillet_radius > 0.:
+            fillet_lower = create_fillet(args, r, lower_fillet_radius, False)
+        if fillet_lower is not None and fillet_upper is not None:
+            fillet = fillet_lower | fillet_upper
+        elif fillet_lower is None:
+            fillet = fillet_upper
+        else:
+            fillet = fillet_lower
+
+        prism = prism & ~fillet
```

Comment:
(Speculation?) If neither fillet is defined then `fillet` will be `None` in this last operation.  Is that geometry operation well defined if `fillet is None`?

If it is.... can you just use the `fillet_upper` and `fillet_lower` without the above logic to see if they are defined?

```suggestion
        prism = prism & ~(fillet_upper | fillet_lower)
```

Are all those operation well defined if one or both are `None`?

Reply:
That's a great question. My guess is that it doesn't work that way based on the documentation for the `|` operator: 
```
class Union(Region, MutableSequence):
     r"""Union of two or more regions.
 
     Instances of Union are generally created via the | operator applied to two
     instances of :class:`openmc.Region`. This is illustrated in the following
     example:
 
     >>> s1 = openmc.ZPlane(z0=0.0)
     >>> s2 = openmc.Sphere(r=637.1e6)
     >>> type(-s2 | +s1)
     <class 'openmc.region.Union'>
 
     Instances of this class behave like a mutable sequence, e.g., they can be
     indexed and have an append() method.
 
     Parameters
     ----------
     nodes : iterable of openmc.Region

```

but maybe @paulromano or @pshriwise can verify?

Reply:
> If neither fillet is defined then `fillet` will be `None` in this last operation. Is that geometry operation well defined if `fillet is None`?

At least one fillet is guaranteed to exist because this code block is contained within an if statement that requires at least one of `uppet_fillet_radius` and `lower_fillet_radius` to be nonzero.